sympy/solvers/bivariate.py

```diff
@@ -374,7 +457,12 @@ def _solve_even_degree_expr(expr, t, symbol):
         if mainexp:
             lhs = collect(lhs, mainexp)
             if lhs.is_Mul and rhs != 0:
-                soln = _lambert(expand_log(log(lhs) - log(rhs)), symbol)
+                if domain.is_subset(S.Reals):
+                    soln = _lambert_real(expand_log(log(lhs) - log(rhs)), symbol)
+                    if soln is S.EmptySet:
+                        return S.EmptySet
+                else:
+                    soln = _lambert(expand_log(log(lhs) - log(rhs)), symbol)
```

Comment:
Since this logic is repeated below. Maybe it should be moved into `_lambert` or something...